Objects/longobject.c

```diff
@@ -50,7 +50,9 @@ get_small_int(sdigit ival)
 {
     assert(IS_SMALL_INT(ival));
     PyObject *v = (PyObject *)&_PyLong_SMALL_INTS[_PY_NSMALLNEGINTS + ival];
+#ifndef Py_IMMORTAL_OBJECTS
     Py_INCREF(v);
+#endif  /* Py_IMMORTAL_OBJECTS */
```

Comment:
We could do this sort of thing in a bunch of places and reduce the overall penalty a bit more, right?

Reply:
Yes! We just need to have a strict guarantee that the incref/decref will only happen on immortal instances.

That just gave me the idea to remove the refcount in `Py_RETURN_TRUE `, `Py_RETURN_FALSE` and `Py_RETURN_NONE `. Just updated the PR to reflect this

Reply:
Awesome!  This is a class of adjustments that @markshannon suggested to me a while back which he believed could buy back 1-2% of a hypothetical 5% penalty.